src/vm/core.zig

```diff
@@ -156,6 +162,23 @@ pub const CairoVM = struct {
             instruction,
             operands_result,
         );
+
+        const OFFSET_BITS: u32 = 16;
+        const off_0 = instruction.off_0 + (@as(i16, 1) << (OFFSET_BITS - 1));
+        const off_1 = instruction.off_1 + (@as(i16, 1) << (OFFSET_BITS - 1));
+        const off_2 = instruction.off_2 + (@as(i16, 1) << (OFFSET_BITS - 1));
+
+        if (self.rc_limits == null) {
+            self.rc_limits = .{ @min(off_0, off_1, off_2), @max(off_0, off_1, off_2) };
+        } else {
+            self.rc_limits = .{ @min(self.rc_limits.?[0], off_0, off_1, off_2), @max(self.rc_limits.?[1], off_0, off_1, off_2) };
+        }
```

Comment:
You can make use of [payload captures](https://ziglearn.org/chapter-1/#payload-captures) here, or `orelse`. if `off_0` is guaranteed we can do a little hack here:

```suggestion
        const limits = self.rc_limit orelse .{ off_0, off_0 };
        self.rc_limits = .{ @min(limits[0], off_0, off_1, off_2), @max(limits[1], off_0, off_1, off_2) };
```

The assumption here is `off_0` will always be around for this